### DISCUSSION TITLE
Order Navigation group

### DISCUSSION THREAD

**Discussion by mstfkhazaal** [filamentphp/filament#6210] [Help] [answered]
I use  
Filament::registerNavigationGroups([
    'Shop',
    'Blog',
    'Settings',
]);
It is work but not work in multi language 
i try this but not work
Filament::registerNavigationGroups([
    __('nav.shop'),
    __('nav.blog'),
     __('nav.setting'),
]);

**Comment by danharrin** [comment]
Try this:

```php
Filament::registerNavigationGroups([
    'Shop' => NavigationGroup::make()->label(__('nav.shop')),
    'Blog' => NavigationGroup::make()->label(__('nav.blog')),
    'Settings' => NavigationGroup::make()->label(__('nav.setting')),
]);
```

Then keep the `$navigationGroup` as English `'Shop'`

**Reply by mstfkhazaal** [reply]
Not work 
![image](https://user-images.githubusercontent.com/47450090/232080319-7b6aa27c-54d4-4d7c-a992-2fe7041f02ce.png)
Provider:
https://github.com/mstfkhazaal/subscribo/blob/master/app/Providers/AppServiceProvider.php#L28

Resources:
https://github.com/mstfkhazaal/subscribo/blob/master/app/Filament/Resources/Options/CurrencyResource.php#L40
https://github.com/mstfkhazaal/subscribo/blob/master/app/Filament/Resources/Options/LocationResource.php#L43

And I'm use Shield Package i cannot change his groupname or any other packages

**Reply by danharrin** [reply]
Try defining `getNavigationGroup()` instead of `$navigationGroup`. In the function you can use `__()`. Then you can register the groups using `Filament::registerNavigationGroups([__('nav.shop'), __('nav.blog'), __('nav.setting')]);`

**Reply by mstfkhazaal** [reply]
I tried it before I asked it's not work

In the correct order in English
![image](https://user-images.githubusercontent.com/47450090/232083495-17010271-61b1-46c9-9296-c22239bb060a.png)
but in other language like arabic
![image](https://user-images.githubusercontent.com/47450090/232083015-e020c9cc-0c8c-40c0-b374-a2d5ec162d91.png)
not same order

I update my project
Provider:
https://github.com/mstfkhazaal/subscribo/blob/master/app/Providers/AppServiceProvider.php#L28

Resources:
https://github.com/mstfkhazaal/subscribo/blob/master/app/Filament/Resources/Options/CurrencyResource.php#L41
https://github.com/mstfkhazaal/subscribo/blob/master/app/Filament/Resources/Options/LocationResource.php#L43

**Reply by danharrin** [reply]
I don't know then, sorry

**Reply by danharrin** [reply]
Actually I have an idea... put the `Filament::registerNavigationGroups()` inside `Filament::serving()`

**Reply by mstfkhazaal** [reply]
Not work 😭

**Reply by danharrin** [reply]
In the middleware array of `config/filament`, make sure your language is being set *before* the ServingFilamentEvent middleware

**Reply by mstfkhazaal** [reply]
I not found ServingFilamentEvent
`config/filament
`
```
    'middleware' => [
        'auth' => [
            Authenticate::class,
        ],
        'base' => [
            EncryptCookies::class,
            AddQueuedCookiesToResponse::class,
            StartSession::class,
            AuthenticateSession::class,
            ShareErrorsFromSession::class,
            VerifyCsrfToken::class,
            SubstituteBindings::class,
            DispatchServingFilamentEvent::class,
            MirrorConfigToSubpackages::class,
        ],
    ],

```
I found my problem
I use session to save my locale Plugin(https://github.com/bezhanSalleh/filament-language-switch)
in Provider the session not start yet app()->getLocale() return default language
It is not possible to access the session from any provider

Do you have any way to access to session in any provider?

I suggest adding groupid 
 protected static ?string $navigationGroupId = 'blog-id';
 
Filament::registerNavigationGroups([
    'shop-id' ,
    'blog-id',
    'settings-id',
]);

**Reply by danharrin** [reply]
The problem is that the middleware in that package is injected after `DispatchServingFilamentEvent`, so that the locale is not set by the time you want to register nav groups

https://github.com/bezhanSalleh/filament-language-switch/blob/main/src/FilamentLanguageSwitchServiceProvider.php#L32-L41

**Reply by mstfkhazaal** [reply]
What should I do?

**Reply by danharrin** [reply]
Open an issue with the package I guess, translating nav groups must be a common request

**Comment by bendeckdavid** [comment, marked as answer]
Hi, this happends because the language middleware from this package is not load yet, you can solve it with a clousure, just like this:

```php
Filament::registerNavigationGroups([
    'Shop' => NavigationGroup::make(fn() => __('nav.shop')),
    'Blog' => NavigationGroup::make(fn() => __('nav.blog')),
    'Settings' => NavigationGroup::make(fn() => __('nav.setting')),
]);
```

**Reply by mstfkhazaal** [reply]
Thank you 

i tried this and it work thanks

 ->navigationGroups([
                'CMS' => NavigationGroup::make(fn() => __('navigation.cms')),
                'Blog' => NavigationGroup::make(fn() => __('navigation.blog')),
                'Settings' => NavigationGroup::make(fn() => __('navigation.settings')),
            ])

**Reply by MasterAboLayan** [reply]
Where should we use this code?

**Reply by TrinsyCa** [reply]
You can use this code in `App\Providers\Filament\DashboardPanelProvider.php` file

```php
return $panel
    ->navigationGroups([
       'Shop' => __('nav.shop'),
       'Blog' => __('nav.blog'),
       'Settings' => __('nav.setting'),
    ])
```

**Comment by christopheraseidl** [comment]
> Hi, this happends because the language middleware from this package is not load yet, you can solve it with a clousure, just like this:
> 
> ```
> Filament::registerNavigationGroups([
>     'Shop' => NavigationGroup::make(fn() => __('nav.shop')),
>     'Blog' => NavigationGroup::make(fn() => __('nav.blog')),
>     'Settings' => NavigationGroup::make(fn() => __('nav.setting')),
> ]);
> ```

This was the solution. Thank you!
